src/mach.rs

```diff
@@ -601,18 +635,19 @@ impl<'a> Mach<'a> {
 }
 
 // FIXME: this should actually return a runtime error if we encounter a from.decl to.decl pair which we don't explicitly match on
-fn build_relocations(artifact: &Artifact, symtab: &SymbolTable) -> Relocations {
+fn build_relocations(segment: &mut SegmentBuilder, artifact: &Artifact, symtab: &SymbolTable) {
     use goblin::mach::relocation::{X86_64_RELOC_BRANCH, X86_64_RELOC_SIGNED, X86_64_RELOC_UNSIGNED, X86_64_RELOC_GOT_LOAD};
-    let mut text_relocations = Vec::new();
-    let mut data_relocations = Vec::new();
+    let text_idx = segment.sections.get_full("__text").unwrap().0;
```

Comment:
in what circumstances would this .unwrap fail? Probably replace with .expect if it is fundamental programmer error

Reply:
It will never fail because `SegmentBuilder::new()` always adds these sections. I don't see the need for using .expect here, but that's a matter of preference. We could also replace these with constant indices.